modules/sor/sorV2/lib/poolsV3/stable/stablePool.ts

```diff
@@ -267,8 +330,16 @@ export class StablePool implements BasePoolV3 {
             scalingFactors: this.tokens.map((t) => t.scalar),
             aggregateSwapFee: 0n,
         };
+
+        if (hookName) {
+            poolState.hookType = hookName;
+        }
```

Comment:
You don't need to add the `if` clause - since both are either `string or undefined`, you should be able to assign one to the other directly:
```suggestion
        poolState.hookType = hookName;
```

Aside from that, I saw this is a requirement from balancer-maths, but why have a `hookName` within `poolState`? Intuitively I'd expect it to be inside `hookState` 😅